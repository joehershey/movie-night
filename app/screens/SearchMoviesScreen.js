import React, { useState } from "react";
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableWithoutFeedback,
  SafeAreaView,
  TextInput,
  ScrollView,
} from "react-native";

import { COLORS, STYLES } from "../assets/saved";
import { FontAwesome5 } from "@expo/vector-icons";

import TabBar from "../components/TabBar";
import TopBar from "../components/TopBar";
import Categories from "../components/Categories";
import Search from "../components/Search";

import fetch from "cross-fetch";

function SearchMoviesScreen(props) {
  const [display, setDisplay] = useState("categories");

  return (
    <SafeAreaView style={STYLES.container}>
      {/* Top Bar */}
      <TopBar
        navigation={props.navigation}
        screen="Landing"
        message="Groups"
      ></TopBar>
      {/* Content */}

      <View style={[STYLES.content, { backgroundColor: COLORS.primary }]}>
        <View
          style={{
            flexDirection: "row",
            justifyContent: "center",
            alignItems: "center",
            margin: 10,
          }}
        >
          <TouchableWithoutFeedback onPress={() => setDisplay("categories")}>
            <View
              style={[
                styles.selectButton,
                display == "categories" ? styles.selected : styles.unselected,
                { borderTopLeftRadius: 10, borderBottomLeftRadius: 10 },
              ]}
            >
              <Text
                style={[
                  {
                    color: "white",
                    fontSize: 20,
                    alignContent: "center",
                  },
                  display == "categories"
                    ? { fontWeight: "bold" }
                    : { fontWeight: "normal" },
                ]}
              >
                Categories
              </Text>
            </View>
          </TouchableWithoutFeedback>

          <TouchableWithoutFeedback onPress={() => setDisplay("search")}>
            <View
              style={[
                styles.selectButton,
                display == "search" ? styles.selected : styles.unselected,
                { borderTopRightRadius: 10, borderBottomRightRadius: 10 },
              ]}
            >
              <Text
                style={[
                  {
                    color: "white",
                    fontSize: 20,
                    alignContent: "center",
                  },
                  display == "search"
                    ? { fontWeight: "bold" }
                    : { fontWeight: "normal" },
                ]}
              >
                Search
              </Text>
            </View>
          </TouchableWithoutFeedback>
        </View>
        {display == "categories" && (
          <Categories
            url={props.url}
            user_id={props.user_id}
            group_id={props.group_id}
            token={props.token}
          ></Categories>
        )}
        {display == "search" && (
          <Search
            url={props.url}
            user_id={props.user_id}
            group_id={props.group_id}
            token={props.token}
          ></Search>
        )}
      </View>

      {/* Tabs */}
      <TabBar navigation={props.navigation} screen="search"></TabBar>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  btn: {
    justifyContent: "center",
    alignItems: "center",
    margin: 5,
    borderRadius: 20,
  },
  selected: {
    backgroundColor: COLORS.secondary,
  },
  selectButton: {
    marginTop: 10,
    marginBottom: 10,
    padding: 10,
  },
  unselected: {
    backgroundColor: "lightgrey",
  },
});
export default SearchMoviesScreen;
