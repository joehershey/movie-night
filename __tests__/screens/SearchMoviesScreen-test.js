import SearchMoviesScreen from '../../app/screens/SearchMoviesScreen';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

describe('Test Search Movies Screen', () => {
    beforeEach( async () => {
        global.tree = renderer.create(<SearchMoviesScreen />).toJSON();
    });

    it("Search movies screen renders properly", async () => {
        await expect(global.tree).toMatchSnapshot();
    });

    it("Does basic test", () => {
        // const toggleShowPopup = jest.fn()
        props = {
        }
        const { getByTestId } = render(<SearchMoviesScreen {...props}/>);
        fireEvent.press(getByTestId('CategoriesButton'));
        fireEvent.press(getByTestId('SearchButton'));
        // expect(getByTestId('MobileCurrentPassword').props.value).toEqual('JESTTestPassword');
    });
});