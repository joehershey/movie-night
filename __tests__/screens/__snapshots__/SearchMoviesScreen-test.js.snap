// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search Movies Screen Search movies screen renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1d1f1e",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "margin": 5,
          }
        }
      >
        Groups
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "paddingBottom": 10,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": 50,
            "marginBottom": 0,
            "width": 100,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flex": 11,
        },
        Object {
          "backgroundColor": "#1d1f1e",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "margin": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "marginBottom": 10,
              "marginTop": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "#166fdb",
            },
            Object {
              "borderBottomLeftRadius": 10,
              "borderTopLeftRadius": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignContent": "center",
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          Categories
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "marginBottom": 10,
              "marginTop": 10,
              "padding": 10,
            },
            Object {
              "backgroundColor": "lightgrey",
            },
            Object {
              "borderBottomRightRadius": 10,
              "borderTopRightRadius": 10,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignContent": "center",
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "fontWeight": "normal",
              },
            ]
          }
        >
          Search
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 11,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "alignItems": "stretch",
            "justifyContent": "flex-start",
          }
        }
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "margin": 10,
                "marginTop": 0,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 20,
                    "justifyContent": "center",
                    "margin": 5,
                  },
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#166fdb",
                    "borderRadius": 2,
                    "padding": 15,
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                  }
                }
              >
                Filters
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "height": 50,
              }
            }
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#1d1f1e",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Schedule
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Queue
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#166fdb",
            "fontWeight": "bold",
            "margin": 5,
          }
        }
      >
        Search
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "normal",
            "margin": 5,
          }
        }
      >
        Settings
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
